astropy_helpers/commands/test.py

```diff
@@ -2,25 +2,33 @@
 Different implementations of the ``./setup.py test`` command depending on
 what's locally available.
 
-If Astropy v1.1.0.dev or later is available it should be possible to import
-AstropyTest from ``astropy.tests.command``.  If ``astropy`` can be imported
-but not ``astropy.tests.command`` (i.e. an older version of Astropy), we can
-use the backwards-compat implementation of the command.
-
-If Astropy can't be imported at all then there is a skeleton implementation
-that allows users to at least discover the ``./setup.py test`` command and
-learn that they need Astropy to run it.
+If Astropy v1.1 or later is available it should be possible to import
+AstropyTest from ``astropy.tests.command``. Otherwise there is a skeleton
+implementation that allows users to at least discover the ``./setup.py test``
+command and learn that they need Astropy to run it.
 """
 
+import os
+from ..utils import import_file
 
 # Previously these except statements caught only ImportErrors, but there are
 # some other obscure exceptional conditions that can occur when importing
 # astropy.tests (at least on older versions) that can cause these imports to
 # fail
+
 try:
-    import astropy  # noqa
-    from astropy.tests.command import AstropyTest
+
+    # If we are testing astropy itself, we need to use import_file to avoid
+    # actually importing astropy (just the file we need).
+    command_file = os.path.join('astropy', 'tests', 'command.py')
+    if os.path.exists(command_file):
```

Comment:
would this work, too when running `astropy.test()` or it doesn't matter at that point anyway as the helpers is not installed?

Reply:
``astropy.test()`` doesn't use the test command (just the test runner directly)